sdk/include/reactos/drivers/videoprt/vpcfgcnv.h

```diff
@@ -0,0 +1,864 @@
+ *              video/monitor configuration data structures.
+ * COPYRIGHT:   Copyright 2023-2024 Hermès Bélusca-Maïto <hermes.belusca-maito@reactos.org>
+ */
+
+/**
+ * @defgroup VpCfgConv  ReactOS Video Port Configuration Data Conversion Helpers
+ *
+ * @brief
+ * Conversion helpers between legacy video/monitor configuration data
+ * structures, and newer ones compatible with CM_PARTIAL_RESOURCE_LIST
+ * resource descriptors.
+ *
+ * RATIONALE:
+ *
+ * The legacy, MIPS-derived, video display controller and monitor
+ * peripheral configuration data is specified, respectively, with
+ * the VIDEO_HARDWARE_CONFIGURATION_DATA (see DDK video.h, with
+ * data starting at the 'Version' field and following) and the
+ * MONITOR_CONFIGURATION_DATA structures (see arc.h), both with
+ * Version == 0 or 1, Revision == 0, as set by the NT OS loader.
+ * They are compatible with the ARC Specification Revision 1.00.
+ *
+ * The newer way to specify configuration data, compatible with
+ * the ARC Specification Revision 1.2, is to use a CM_PARTIAL_RESOURCE_LIST
+ * with one or more CM_PARTIAL_RESOURCE_DESCRIPTOR's for enumerating
+ * standard resources (Interrupts, I/O ports, Memory...) as well
+ * as free-form device-specific information data.
+ *
+ * For video display controllers, all the legacy data can be
+ * described by a set of standard CM_PARTIAL_RESOURCE_DESCRIPTOR's:
+ * - The Irql/Vector interrupt settings are now provided with
+ *   a CmResourceTypeInterrupt descriptor.
+ * - Both ControlBase/Size and CursorBase/Size memory I/O ports
+ *   are provided with successive CmResourceTypePort descriptors,
+ *   *respectively in this order*.
+ * - The framebuffer's FrameBase/Size is provided, in a 64-bit
+ *   compatible way, with a CmResourceTypeMemory descriptor.
+ *
+ * Any other extended video controller-specific data, that would
+ * have been described with a vendor-specific extended version of
+ * the VIDEO_HARDWARE_CONFIGURATION_DATA structure, can now be
+ * specified with a CmResourceTypeDeviceSpecific descriptor appended
+ * to the CM_PARTIAL_RESOURCE_LIST.
+ * For example, in ReactOS, extended framebuffer format of the
+ * boot console may be specified with the ReactOS-specific
+ * CM_FRAMEBUF_DEVICE_DATA structure (see framebuf.h).
+ *
+ * For monitor peripherals, all the legacy data is extended by
+ * the CM_MONITOR_DEVICE_DATA structure (see DDK ntddk.h or wdm.h),
+ * to be specified with a CmResourceTypeDeviceSpecific descriptor.
+ *
+ * The ReactOS loader specifies this configuration data in the
+ * new format. In order to ensure backward compatibility with
+ * _legacy_ Windows video miniports that could use this data
+ * (typically on NT <= 4 or 5, mostly for MIPS-based machines
+ * but also few x86 ones), and for interoperability of our code
+ * with the Windows NT loader, we define a set of conversion
+ * functions that can translate between legacy (ARC Rev.1.00)
+ * and newer (ARC Rev.1.2) configuration data structures.
```

Comment:
might make sense to place this at the top of the RATIONALE section. I had to read through quite a bit until the abstract question in my head about "the why" was answered here at the very end.